ServerNG.md

```diff
@@ -0,0 +1,155 @@
+
+The server configuration extends the standard Infinispan configuration adding server-specific elements:
+
+* `security` configures the available security realms which can be used by the endpoints.
+* `cache-container` multiple containers may be configured, distinguished by name.
+* `endpoints` lists the enabled endpoint connectors (hotrod, rest, ...).
+* `socket-bindings` lists the socket bindings.
+
+An example skeleton configuration file looks as follows:
+
+```
+<infinispan>
+
+   <!-- Global security configuration -->
+   <security>
+      <security-realm name="ManagementRealm">
+         <authentication>
+            <properties path="mgmt-users.properties" relative-to="infinispan.server.config.dir"/>
+         </authentication>
+         <authorization map-groups-to-roles="false">
+            <properties path="mgmt-groups.properties" relative-to="infinispan.server.config.dir"/>
+         </authorization>
+      </security-realm>
+      <security-realm name="PublicRealm">
+         <server-identities>
+            <ssl>
+                <keystore path="application.keystore" relative-to="infinispan.server.config.dir" keystore-password="password" alias="server" key-password="password" generate-self-signed-certificate-host="localhost"/>
+            </ssl>
+         </server-identities>
+         <authentication>
+            <properties path="application-users.properties" relative-to="infinispan.server.config.dir"/>
+         </authentication>
+         <authorization>
+            <properties path="application-roles.properties" relative-to="infinispan.server.config.dir"/>
+         </authorization>
+      </security-realm>
+   </security>
+   
+   <!-- JGroups configuration -->
+   <jgroups transport="org.infinispan.remoting.transport.jgroups.JGroupsTransport">
+      <stack-file name="udp" path="jgroups-udp.xml"/>
+      <stack-file name="tcp" path="jgroups-tcp.xml"/>
+   </jgroups>
+   
+   <!-- Cache containers -->
+   <cache-container name="default" ...>
+      <transport .../>
+   </cache-container>
+   
+   <!-- Endpoints -->
+   <endpoints>
+      <hotrod-connector cache-container="default" socket-binding="hotrod" ... />
+      <rest-connector socket-binding="rest" ... />
+      <admin-connector socket-binding="management" security-realm="ManagementRealm" />
+   </endpoints>
+   
+   <socket-bindings>
+      <socket-binding name="hotrod" address="${infinispan.bind.address}" port="11222"/>
+      <socket-binding name="rest" address="${infinispan.bind.address}" port="8080"/>
+      <socket-binding name="management" address="${infinispan.bind.address.management}" port="9990"/>
+      <socket-binding name="cluster" multicast-address="${infinispan.bind.address.multicast}" multicast-port="45700"/>
+   </socket-bindings>
+</infinispan>
+```
+
+# Logging
+
+Logging is handled by JBoss Logging's LogManager. This is configured through a `logging.properties` file in the 
+`server/configuration` directory. The following is an example:
+
```

Comment:
use a properties file is unlikely, it would be better to have a xml configuration for this as it is for wildfly.
Maybe it can be a choice which prefer xml?

Reply:
I wanted to minimize the effort of parsing yet another custom structure and just delegating it to something external. Also, logging != configuration in my opinion.